dspace-server-webapp/src/test/java/org/dspace/app/rest/EPersonRegistrationRestControllerIT.java

```diff
@@ -0,0 +1,343 @@
+                          .withCanLogin(true)
+                          .build();
+
+        context.restoreAuthSystemState();
+    }
+
+
+    @Test
+    public void givenOrcidToken_whenPostForMerge_thenUnauthorized() throws Exception {
+
+        getClient().perform(
+            post("/api/eperson/epersons/" + customEPerson.getID())
+                .param("token", orcidRegistration.getToken())
+                .param("override", "eperson.firtname,eperson.lastname,eperson.orcid")
+        ).andExpect(status().isUnauthorized());
+
+    }
+
+    @Test
+    public void givenExpiredToken_whenPostForMerge_thenUnauthorized() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+        registrationDataService.markAsExpired(context, orcidRegistration);
+        context.restoreAuthSystemState();
+
+        getClient().perform(
+            post("/api/eperson/epersons/" + customEPerson.getID())
+                .param("token", orcidRegistration.getToken())
+                .param("override", "eperson.firtname,eperson.lastname,eperson.orcid")
+        ).andExpect(status().isUnauthorized());
+
+    }
+
+    @Test
+    public void givenExpiredToken_whenPostAuthForMerge_thenForbidden() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+        registrationDataService.markAsExpired(context, orcidRegistration);
+        context.restoreAuthSystemState();
+
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+
+        getClient(tokenAdmin).perform(
+            post("/api/eperson/epersons/" + customEPerson.getID())
+                .param("token", orcidRegistration.getToken())
+                .param("override", "eperson.firtname,eperson.lastname,eperson.orcid")
+        ).andExpect(status().isForbidden());
+
+    }
+
+    @Test
+    public void givenValidationRegistration_whenPostAuthDiffersFromIdPathParam_thenForbidden() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+        RegistrationData validationRegistration =
+            registrationDataService.create(context, "0000-0000-0000-0000", RegistrationTypeEnum.VALIDATION_ORCID);
+        context.restoreAuthSystemState();
+
+        String tokenAdmin = getAuthToken(admin.getEmail(), password);
+
+        getClient(tokenAdmin).perform(
+            post("/api/eperson/epersons/" + customEPerson.getID())
+                .param("token", validationRegistration.getToken())
+        ).andExpect(status().isForbidden());
+
+    }
+
+    @Test
+    public void givenValidationRegistration_whenPostWithoutOverride_thenCreated() throws Exception {
+
+        Email spy = Mockito.spy(Email.class);
+        doNothing().when(spy).send();
+
+        emailMockedStatic.when(() -> Email.getEmail(any())).thenReturn(spy);
+
+        context.turnOffAuthorisationSystem();
+        RegistrationDataChanges changes =
+            new RegistrationDataChanges("vincenzo.mecca@4science.com", RegistrationTypeEnum.VALIDATION_ORCID);
```

Comment:
Just a note, you might not want your real email address in the codebase.  We usually use fake emails in tests.  This real email appears several times in this IT class.